docs/en/reference/testing.rst

```diff
@@ -0,0 +1,95 @@
+Testing Guidelines
+===================
+
+In order to ensure high quality, all components of the Doctrine DBAL library are extensively covered with tests.
+
+Having the code covered with tests and running all tests against each individual code change helps prevent
+breakages of the library logic when its code changes.
+
+Additionally, tests:
+
+1. Help maintainers understand what problem the given code change is trying to solve.
+2. Make sure that the problem being solved needs to be solved in the DBAL.
+3. Document the proper usage of the DBAL APIs.
+
+Requirements
+------------
+
+1. Each pull request that adds new logic or changes the existing logic must have tests.
+
+    (!) Exceptions to the rule:
```

Comment:
This is meant visually highlighted (I'll figure out the RST syntax for this later).